activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -67,11 +67,21 @@ def query(sql, name = nil) #:nodoc:
           end
         end
 
+        READ_QUERY = ActiveRecord::ConnectionAdapters::AbstractAdapter.build_read_query_regexp.call(:select, :show, :set) # :nodoc:
```

Comment:
`explain` ?

Reply:
Explain is on master. https://github.com/rails/rails/blob/72005ad0e13cb83d0e5ed7f9af7e0537efcf8963/activerecord/lib/active_record/connection_adapters/abstract_adapter.rb#L66